packages/yarnpkg-core/sources/Project.ts

```diff
@@ -1838,7 +1835,12 @@ export class Project {
     }
   }
 
-  async cacheCleanup({cache, report}: InstallOptions)  {
+  async cacheCleanup({cache, mode, persistProject, report}: InstallOptions)  {
+    if (mode === InstallMode.UpdateLockfile)
```

Comment:
tbh I stared at this for a while trying to invert the guard clause. I think I got it right and am expressing the semantics clearly here, but please double check my work.